request.go

```diff
@@ -131,14 +131,28 @@ func unmarshalNode(data *Node, model reflect.Value, included *map[string]*Node)
 
 	for i := 0; i < modelValue.NumField(); i++ {
 		fieldType := modelType.Field(i)
-		tag := fieldType.Tag.Get("jsonapi")
+		tag := fieldType.Tag.Get(annotationJSONAPI)
+
+		// handles embedded structs
+		if isEmbeddedStruct(fieldType) {
+			if shouldIgnoreField(tag) {
+				continue
+			}
+			model := reflect.ValueOf(modelValue.Field(i).Addr().Interface())
```

Comment:
I think whether I assign to the original argument, or shadow it here, it shouldn't make a difference since the main loop of this function works off of `modelValue`.

Between assign/overwriting to the original argument vs initializing a new var, I feel like the latter is preferred.  But to avoid confusion, I'll pass it directly to `unmarshalNode()` vs using a shadowed var.

Also, note, w/ handling `annotationRelation`, we don't overwrite the original `model` argument.   https://github.com/google/jsonapi/blob/master/request.go#L474